### ISSUE TITLE
Algorithms that require scaling

### ISSUE THREAD

**Issue by slabko** [ageron/handson-ml#26]
Hello, Aurélien. 

Thank you a lot for such a great and comprehensive book. 

I have one question regarding exercises in chapter 4. 

> Question: Suppose the features in your training set have very different scales. What algorithms might suffer from this, and how? What can you do about it?
> 
> Answer (from Excercise solutions): If the features in your training set have very different scales, the cost function will have the shape of an elongated bowl, so the Gradient Descent algorithms will take a long time to converge. To solve this you should scale the data before training the model. Note that the Normal Equation will work just fine without scaling. 

I understand that actually in the case of Gradient Descent, given enough time it converges anyway (although it might hit the limits on the number of iterations). However, with regularization models, the situation is even worse, because, in the case of different feature scales, the model can be less precise. Looking the documentation I found that LogisticRegression also uses some regularization and therefore can suffer noticeably.

That was my conclusion, but I didn't find in the exercise solution and thought that I might miss something.

**Comment by ageron** [comment]
Hi Andrew,
Thanks for your kind words. :)
That's an excellent point: if a model is regularized, then scaling is even more important. I'll add it to the solution.

**Comment by ageron** [comment]
FYI, I added the following sentence at the end of this exercise's solution: "Moreover, regularized models may converge to a suboptimal solution if the features are not scaled: indeed, since regularization penalizes large weights, features with smaller values will tend to be ignored compared to features with larger values."
